DNS suffix customization

Mook-as opened the pull request:
There was some short [discussion] on #integration, but I'm certainly up for more if that's desired.  This is basically somewhere that lets me keep track of what's going on :)

This allows changing the DNS suffix used to look for other etcd nodes (instead of consul's `etcd.service.cf.internal`), for cases where a different service discovery mechanism is being used.  If this lands as-is, the next step is to update manifest generation in cf-release and diego-release (anything else?) to add the property, then remove the default from the spec.

(I can't quite tell from the git history if I should be working from `develop` or `master`; the README wasn't very enlightening on that aspect.  I'd be happy to rebase or whatever else to make the code sensible, first.)

If opening a story in a tracker to coordinate dealing with `cf-release` / `diego-release` is preferable, that's fine too, though it would be a bit harder for me to help out :)

[discussion]: https://cloudfoundry.slack.com/archives/infrastructure/p1456964046000559

Amit-PivotalLabs commented on jobs/etcd/templates/etcd_bosh_utils.sh.erb:
```diff
@@ -27,11 +27,14 @@ node_name="<%= name.gsub('_', '-') %>-<%= spec.index %>"
   peer_protocol = p("etcd.peer_require_ssl") ? "https" : "http"
 
+  dns_suffix = p("etcd.dns_suffix")
+  dns_suffix = ".#{dns_suffix}" unless dns_suffix.empty?
```
Could you explain this part? When would you allow `dns_suffix` to be empty?

Mook-as replied:
When your DNS search suffix does the right thing, mostly (i.e. `search example.com` in `/etc/resolv.conf`).  In our case, this is a local dev VM thing (that isn't ready for the world, and might cease to exist before that happens) where that's the case.

Not too concerned if we don't want that to be legal (can always change the VM), but it seemed good to correctly support somebody explicitly giving an empty string.

Mook-as commented:
Updated PR per comments on Slack:
- Changed the property name to `etcd.advertise_urls_dns_suffix`
- Remove the comment about deprecation
- Drop handling empty strings (i.e. no suffix given).

Amit-PivotalLabs commented:
Thanks, prioritized.
